projects/components/src/download-file/service/file-download.service.ts

```diff
@@ -31,11 +31,14 @@ export class FileDownloadService {
    * @param config Csv download config
    */
   public downloadAsCsv(config: CsvDownloadFileConfig): Observable<FileDownloadEvent> {
+    const getAllDataKeys = (data: Dictionary<unknown>[]): string[] => uniq(data.map(row => Object.keys(row)).flat());
     // If given header is empty, then create header from the data keys
     const header$ = isEmpty(config.header)
       ? config.dataSource.pipe(
-          map(data => Object.keys(data[0] ?? [])),
-          map(keys => keys.map(startCase))
+          map(
```

Comment:
give there's a decent amount of logic here, can we also write a real test?

Reply:
bump here.

Reply:
+1 one the test

Reply:
Added: [`6b4d3d1` (#2478)](https://github.com/hypertrace/hypertrace-ui/pull/2478/commits/6b4d3d18f8937d8d8deebb2422e1a4610bbfc6d5)